packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -266,30 +280,51 @@ export class UploadComponent implements OnInit {
         ),
       })
         .pipe(
-          tap((joined_audio_and_ras: any) => {if (joined_audio_and_ras.readalong instanceof HttpErrorResponse) {
+          catchError((err, caught) => {
             this.loading = false;
-          } }),
-          takeWhile((joined_audio_and_ras: any) => !(joined_audio_and_ras.readalong instanceof HttpErrorResponse)),
+            throw err;
+          }),
```

Comment:
Why remove takeWhile? We should unsubscribe from the observable when it throws an error.

Reply:
If the observable throws an error, it stops, and reports it to the error callback, no?

Reply:
(I agree that checking the error type below is not so great, but it is at least semantically equivalent to a series of catch blocks for each type of exception)